tools/diff-processor/detector/detector.go

```diff
@@ -18,9 +19,44 @@ type MissingTestInfo struct {
 
 type FieldSet map[string]struct{}
 
+// ResourceChanges is a nested map with field names as keys and Field objects
+// as bottom-level values.
+// Fields are assumed not to be covered until detected in a test.
+type ResourceChanges map[string]any
```

Comment:
```suggestion
type ResourceChanges map[string]Field
```

Will the bottom-level value always be a field?